dashboard/app/controllers/student_work_sample_controller.rb

```diff
@@ -89,14 +89,13 @@ def fetch_student_code_samples
   end
 
   def fetch_student_code_samples_without_evaluations(level, unit_id, num_samples)
-    # We want to pull samples from students who have been assigned to work on the level.
-    sections = Section.where(script_id: unit_id)
-    student_ids = Follower.where(section: sections).pluck(:student_user_id)
+    # Find users who have worked on this level.
+    student_ids = UserLevel.where(level_id: level.id, script_id: unit_id).pluck(:user_id)
```

Comment:
@kobryan0619 made a great point on an earlier review that because we're looking at historical data the most recently assigned unit might not be CSP Unit 4; for example if the section completed all of CSP. This widens our sampling pool.